src/Features/Core/Portable/MetadataAsSource/IMetadataAsSourceFileProvider.cs

```diff
@@ -16,30 +12,19 @@ internal interface IMetadataAsSourceFileProvider
     /// <summary>
     /// Generates a file from metadata. Will be called under a lock to prevent concurrent access.
     /// </summary>
-    Task<MetadataAsSourceFile?> GetGeneratedFileAsync(
+    Task<(MetadataAsSourceFile file, MetadataAsSourceFileMetadata fileMetadata)?> GetGeneratedFileAsync(
```

Comment:
What would it look like if we have a MetadataAsSourceFileWithMetadata which holds the first one rather than having the tuple? There was one place I flagged where there was a bunch of null suppressions happening and maybe it'd clean that up?